### PR TITLE
build ubuntu 22.04 image

### PR DESCRIPTION
**Pull request by lsm5**
Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

### CODE REVIEW

**Comment by lsm5** [comment]
We should move away from the kubic repos. Either switch to default repos or use my shiny fedora-spec generated kubic repos.

**Comment by lsm5** [comment]
enabled the unstable repo which uses fedora generated packages.

File: base_images/gce.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ variables:  # Empty value means it must be passed in on command-line
 
     # Upstream source for Ubuntu image to duplicate (prevents expiration).
     # Use the most recent image based on this family name.
-    UBUNTU_BASE_FAMILY: 'ubuntu-2110'
+    UBUNTU_BASE_FAMILY: 'ubuntu-2204'
```
[the remark this answers is not in the document]
Comment (cevich) [reply base_images/gce.yml:26]: yes it's possible, that's something on my bucket list, but sheer laziness always wins for some reason lol :laughing:  Same thing for "fedora" and "prior-fedora".  I'll open an issue on it so I don't keep forgetting :wink:

[the remark this answers is not in the document]
Comment (cevich) [reply base_images/gce.yml:26]: https://github.com/containers/automation_images/issues/135

**Comment by cevich** [comment]
This failure is strange [in the container build](https://cirrus-ci.com/task/6010356995719168?logs=main#L1016): 

```
rm: cannot remove '/var/cache/apt': Device or resource busy
```

But the ultimate failure is a missing `golang-github-containers-common`.  I'm guessing that's something that came from kubik before?  -sigh-

**Comment by cevich** [comment]
This is going to barf with a 404, need to replace with URLs under the `release` tree:

```
    # Latest Fedora release (qcow and CHECKSUM) download URLs
    # N/B: There are Fedora-Cloud...GCP.tar.gz images available, however
    #      as of this comment, they lack the cloud-init package which is
    #      required by GCP for startup resizing of the rootfs.
    FEDORA_CSUM_URL: "https://dl.fedoraproject.org/pub/fedora/linux/development/36/Cloud/x86_64/images/Fedora-Cloud-36-x86_64-20220428.n.0-CHECKSUM"
    FEDORA_IMAGE_URL: "https://dl.fedoraproject.org/pub/fedora/linux/development/36/Cloud/x86_64/images/Fedora-Cloud-Base-36-20220428.n.0.x86_64.qcow2"
```

**Review by cevich — APPROVED** [review]
Looks good so far.

**Comment by lsm5** [comment]
> ```
> rm: cannot remove '/var/cache/apt': Device or resource busy
> ```

i'm fixing that one. Once the fedora-derived debs are built, that will go away.

**Comment by lsm5** [comment]
> https://dl.fedoraproject.org/pub/fedora/linux/development/36/Cloud/x86_64/images/

done

**Comment by lsm5** [comment]
> > ```
> > rm: cannot remove '/var/cache/apt': Device or resource busy
> > ```
> 
> i'm fixing that one. Once the fedora-derived debs are built, that will go away.

alrighty, buildah and skopeo should now be available on that repo so that's 1 issue less.

**Comment by cevich** [comment]
`make: *** No rule to make target 'install.libseccomp.sudo'.  Stop.`

Looks like that target was removed from podman's Makefile https://github.com/containers/podman/commit/d8c42de4121f5c2088db5a492ce600d0b99f8e75

**Comment by cevich** [comment]
If you want to include the removed podman install into `cache_images/podman_tooling.sh` for Ubuntu, I'm okay with that.  Though I don't know what it does, if it's still needed, or if that's the right fix :confounded:

**Comment by lsm5** [comment]
@cevich 
```
==> Some builds didn't complete successfully and had errors:
--> ubuntu: Error getting source image for instance creation: Could not find image, ubuntu-2204, in projects, [ubuntu-os-cloud]: 1 error(s) occurred:
```

should the image be called ubuntu-2204 or ubuntu-2204-lts or something else?

**Comment by lsm5** [comment]
@cevich ready to go :crossed_fingers:

**Comment by lsm5** [comment]
@cevich added rngd to ubuntu base setup. PTAL.

File: base_images/ubuntu_base-setup.sh [hunk 2]
```diff
@@ -24,9 +24,14 @@ PKGS=( \
     git
     openssh-client
     openssh-server
+    rng-tools5
     software-properties-common
 )
 
+if ! ((CONTAINER)); then
+    systemctl enable rngd
```
Comment (cevich) [inline base_images/ubuntu_base-setup.sh:32]:
```suggestion
    $SUDO systemctl enable rngd
```

**Comment by cevich** [comment]
@lsm5 I also found a bug with the `podman_tooling.sh` script, sending up a PR...

**Comment by cevich** [comment]
...https://github.com/containers/automation_images/pull/136 (assuming I didn't break things) it would be nice to have that in here.

**Comment by cevich** [comment]
I checked, it's [called `rngd.service`](https://cirrus-ci.com/task/5794812753346560?logs=main#L77), so something else is going on...

File: base_images/ubuntu_base-setup.sh [hunk 3]
```diff
@@ -24,9 +24,14 @@ PKGS=( \
...
     openssh-client
     openssh-server
+    rng-tools5
     software-properties-common
 )
 
+if ! ((CONTAINER)); then
+    $SUDO systemctl enable rngd
+fi
+
```
Comment (cevich) [inline base_images/ubuntu_base-setup.sh:34]: This section needs to move down below `$SUDO apt-get -qq -y install "${PKGS[@]}"`

Comment (lsm5) [reply base_images/ubuntu_base-setup.sh]: PTAL.

**Comment by cevich** [comment]
I think the word "successful" speak for it's self :grinning: Nice job.

**Pull request merged**